### PR TITLE
working version

### PR DESCRIPTION
**Pull request by odinuv**
https://keboola.atlassian.net/browse/PS-1169

### CODE REVIEW

File: src/ClientWrapper.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        $this->branchId = $branchId;
+    }
+
+    public function getBasicClient()
+    {
+        return $this->client;
+    }
+
+    public function getBranchClient()
```
Comment (MiroCillik) [inline src/ClientWrapper.php:37]: Nemozu tu vznikat podivne stavy? 
Zavolam: 
```
$client->setBranch('miro');
$miroBranchClient = $client->getBranchClient();
$client->setBranch('odin');
$odinBranchClient = $client->getBranchClient();
```
Ale v oboch branch clientoch ($miroBranchClient a $odinBranchClient) bude nastavena branch `miro` ak sa nemylim.

Comment (MiroCillik) [reply src/ClientWrapper.php:37]: Chapem, ze chceme mat moznost prepnut branch a zaroven udrzovat instanciu toho branchClienta. Aj ked to si nie som isty, ze je potreba.
Nemoze to fungovat ako factory a proste to vzdy vratit novu instanciu? Tzn wrapper si nebude udrzovat instanciu branchClienta, ale metoda getBranchClient vzdy vrati novu instanciu s aktualnym branchId.

Comment (MiroCillik) [reply src/ClientWrapper.php:37]: Alebo v metode `setBranch()` nastavit `$this->branchClient = null;`

Comment (odinuv) [reply src/ClientWrapper.php:37]: no mas naprostou pravdu - cely to souvisi tady s touhle vypicenosti - klienta potrebuju vytvorit tady https://github.com/keboola/docker-bundle/pull/514/files#diff-0ca310e8666080e00e971a61e5585099a616c008f66c6cf2f54bba884a57151aR110 a branch id zjistim az tady https://github.com/keboola/docker-bundle/pull/514/files#diff-0ca310e8666080e00e971a61e5585099a616c008f66c6cf2f54bba884a57151aR182

Asi bych mel tendenci rict, ze pokud zavolas setBranch a je uz branch nastaveny, tak to hodi exception, protoze v ramci jednoho jobu by se branch vubec nemela menit, slo by to?

Comment (MiroCillik) [reply src/ClientWrapper.php:37]: Jo, jasne, tahleta vypicenost... Uz je mi to jasne.

Comment (MiroCillik) [reply src/ClientWrapper.php:37]: JJ, to riesenie co navrhujes je OK.

**Review by MiroCillik — APPROVED** [review]
OK

**Pull request merged**